Assets/Src/AuthController/Exceptions/NotSelectedFirebaseCredentialProviderException.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+
+namespace Src.AuthController.Exceptions
+{
+    [Serializable]
+    public class NotSelectedFirebaseCredentialProviderException : Exception
```

Comment:
"NotSelected" sounds like the user or the client code did not provide something or rather did not select something.  Consider using word "Unfamiliar" for cases like that e.g. UnfamiliarFirebaseCredentialProviderException. Such name will cary more clear information about the situation in which such exceptions are thrown.